src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,78 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+	public delegate void RawGamepadCallback (RawGamepad raw_gamepad);
```

Comment:
We are defining RawGamepadCallback 3 times, maybe it should be in the LibGamepad namespace and its own file directly (same for GamepadCallback).